evap/static/js/datagrid.js

```diff
@@ -0,0 +1,314 @@
+                }
+            }
+            return 0;
+        });
+    }
+
+    // Reflects changes to the rows to the DOM
+    update() {
+        this.container.children().detach();
+        const elements = this.rows
+            .filter(row => row.isDisplayed)
+            .map(row => row.element);
+        this.container.append($(elements));
+        this.saveToStorage();
+    }
+
+    loadFromStorage() {
+        this.state = Object.assign({
+            search: "",
+            filter: {},
+            order: this.defaultOrder,
+        }, JSON.parse(localStorage.getItem(this.storageKey)));
+        this.syncFilterStateWithInputs();
+        this.filterRows();
+        this.sortRows();
+        this.update();
+    }
+
+    saveToStorage() {
+        localStorage.setItem(this.storageKey, JSON.stringify(this.state));
+    }
+
+    syncFilterStateWithInputs() {
+        this.searchInput.val(this.state.search);
+    }
+}
+
+// Table based data grid which uses its head and body
+export class TableGrid extends DataGrid {
+    init({table, ...options}) {
+        this.sortableHeaders = new Map();
+        table.find("thead").find("th.col-sorting").each((index, header) => {
+            const position = $(header).index();
+            this.sortableHeaders.set(position, $(header));
+        });
+        super.init({container: table.find("tbody"), ...options});
+    }
+
+    findSearchableCells(row) {
+        return row.children().get();
+    }
+
+    addFilterRowData(row, data) {}
+
+    findCellByColumn(row, column) {
+        return row.children().eq(column);
+    }
+
+    get defaultOrder() {
+        if (this.sortableColumns.length > 0) {
+            return [[this.sortableColumns[0], "asc"]];
+        } else {
+            return [];
+        }
+    }
+}
+
+export class EvaluationGrid extends TableGrid {
+    init({filterButtons, ...options}) {
+        this.filterButtons = filterButtons;
+        super.init(options);
+    }
+
+    bindEvents() {
+        super.bindEvents();
+        this.filterButtons.each((index, element) => {
+            const button = $(element);
+            const count = this.rows.filter(row => {
+                return row.filterValues.get("state").includes(button.data("filter"));
+            }).length;
+            const badgeClass = count === 0 ? "badge-btn-zero" : "badge-btn";
+            button.append(`<span class="badge badge-pill ${badgeClass}">${count}</span>`);
+
+            button.click(() => {
+                if (button.hasClass("active")) {
+                    button.removeClass("active");
+                    this.filter({state: []});
+                } else {
+                    button.addClass("active");
+                    this.filterButtons.not(button).removeClass("active");
+                    this.filter({state: [button.data("filter")]});
+                }
+            });
+        });
+    }
+
+    addFilterRowData(row, data) {
+        const state = row.find(".icon").get()
```

Comment:
I thought I had seen you replacing the icon matching by something else, or do i misremember?